### PR TITLE
Migration to TF2 and addition of time-series architectures

### PR DESCRIPTION
**Pull request by devisperessutti**
This MR adds the following features:

 * migration to TF2, using `Keras` API to facilitate creation, training and testing of deep learning architectures;
 * re-structuring of modules into more meaningful categories;
 * addition of new architectures for time-series classification (e.g. `TCN`, `TempCNN`, `RNN` and `TransformerEncoder`);
 * support for weighted loss function, and focal loss for semantic segmentation;
 * general improvements to code styling.

### CODE REVIEW

File: eoflow/models/segmentation_unets.py [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+#             if model_head == ModelHeads.TRAIN:
+#                 heads.append(train_head)
+#             elif model_head == ModelHeads.PREDICT:
+#                 heads.append(predict_head)
+#             elif model_head == ModelHeads.EVALUATE:
+#                 heads.append(evaluate_head)
+#             else:
+#                 raise NotImplementedError
+#
+#         return heads
```
Comment (lojzezust) [inline eoflow/models/segmentation_unets.py]: Should we remove this from the code before merging? We can add it back once we have converted the code.

File: eoflow/models/layers.py [hunk 2]
```diff
@@ -1,338 +1,304 @@
...
+    def call(self, inputs, training=None):
+        r = self.layer(inputs, training=training)
+
+        # Squeeze along temporal dimension
+        return tf.squeeze(r, axis=[1])
+
+
+# OLD, NOT YET CONVERTED
 
 def conv2d_gru(input_, nfeats_out, k_size=3, scope='reduce_t', padding='VALID', return_sequence=False):
```
Comment (lojzezust) [inline eoflow/models/layers.py]: I think this should also be removed before merging. It's leftover code, not really useful without converting.

File: eoflow/models/conv_cells.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 """ Implementation of ConvLSTM and ConvGRU derived from https://github.com/carlthome/tensorflow-convlstm-cell """
```
Comment (lojzezust) [inline eoflow/models/conv_cells.py]: This entire file is also deprecated and should be removed. If we update it we can include it later.
